Maple2.Server.Game/Manager/ExperienceManager.cs

```diff
@@ -57,18 +60,76 @@ public sealed class ExperienceManager {
     }
 
     private long GetRestExp(long expGained) {
-        long addedRestExp = Math.Min(RestExp, (long)(expGained * (Constant.RestExpAcquireRate / 10000.0f))); // convert int to a percentage
+        long addedRestExp = Math.Min(RestExp, (long) (expGained * (Constant.RestExpAcquireRate / 10000.0f))); // convert int to a percentage
         RestExp = Math.Max(0, RestExp - addedRestExp);
         Exp += expGained;
         return addedRestExp;
     }
 
     public void AddExp(long expGained, ExpMessageCode expMessageCode = ExpMessageCode.s_msg_take_exp) {
         expGained += GetRestExp(expGained);
-        LevelUp();
+        if (LevelUp()) {
+            session.PlayerInfo.SendUpdate(new PlayerUpdateRequest {
```

Comment:
why not move this into the `LevelUp()` function since it's more related to `LevelUp` than `AddExp`?